### PR TITLE
Use service connection for MAR ingestion and use .default scope

### PR DESCRIPTION
**Pull request by lbussell**
This should fix https://github.com/dotnet/docker-tools/issues/1286 once the new ImageBuilder is built and flows in.

Using the service connection by itself yielded a new error:

```
 ---> Azure.Identity.AuthenticationFailedException: ClientAssertionCredential authentication failed: AADSTS1002012: The provided value for scope <snip> is not valid. Client credential flows must have a scope value with /.default suffixed to the resource identifier (application ID URI).
```

So that's why the ImageBuilder change is necessary. I will file a follow-up issue to use a pipeline variable for the status API scope since hardcoding it isn't great.

### CODE REVIEW

File: eng/common/templates/steps/wait-for-mcr-image-ingestion.yml [hunk 1]
```diff
@@ -9,12 +9,13 @@ steps:
...
     displayName: Wait for Image Ingestion
     condition: and(${{ parameters.condition }}, eq(variables['waitForIngestionEnabled'], 'true'))
+    serviceConnection: $(marStatus.serviceConnectionName)
+    internalProjectName: 'internal'
     args: >
       waitForMcrImageIngestion
       '${{ parameters.imageInfoPath }}'
       --manifest '$(manifest)'
       --repo-prefix '$(publishRepoPrefix)'
-      --min-queue-time '${{ parameters.minQueueTime }}'
```
Comment (mthalman) [inline eng/common/templates/steps/wait-for-mcr-image-ingestion.yml:17]: Why is this removed?

**Comment by lbussell** [comment]
Filed https://github.com/dotnet/docker-tools/issues/1288

File: eng/common/templates/steps/wait-for-mcr-image-ingestion.yml (hunk 1 above)
Comment (lbussell) [reply eng/common/templates/steps/wait-for-mcr-image-ingestion.yml:17]: For testing. I will revert this change.

**Review by mthalman — APPROVED** [review]
